.github/workflows/validate-all.yml

```diff
@@ -16,53 +16,92 @@ on:
-          git diff-tree --no-commit-id --name-only -r "$parent" ${{ github.sha }} | tee "$OUTFILE"
-        fi
-    - name: Run validation
-      run: |
-        if [ ${{ github.event_name }} == "schedule" ]; then
-          echo "Validating all projects (schedule)"
-          python validate.py -v
-        else
-          export FILELIST=$HOME/filelist.txt
-          cat "$FILELIST"
-          if grep -q '^validate.py' "$FILELIST"; then
-            echo "Validating all projects (script changed)"
+      - name: Run validation
+        run: |
+          if [ ${{ github.event_name }} == "schedule" ]; then
+            echo "Validating all projects (schedule)"
             python validate.py -v
           else
-            projects=$(cat "$FILELIST" | grep '^annotated-projects' | cut -d/ -f 2)
-            echo "Validating projects:" $projects
-            python validate.py -v $projects
+            export FILELIST=$HOME/filelist.txt
+            cat "$FILELIST"
+            if grep -q '^validate.py' "$FILELIST"; then
+              echo "Validating all projects (script changed)"
+              python validate.py -v
+            else
+              projects=$(cat "$FILELIST" | grep '^annotated-projects' | cut -d/ -f 2)
+              echo "Validating projects:" $projects
+              python validate.py -v $projects
+            fi
           fi
-        fi
+  mage_tab_validation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup conda
+        uses: s-weigand/setup-conda@v1
+        with:
+          update-conda: true
+          conda-channels: anaconda, bioconda, conda-forge
+      - name: Install perl-atlas-modules
+        run: conda create -n perl-atlas -c ebi-gene-expression-group -c bioconda perl-atlas-modules perl-graphviz
+      - name: Validate General MAGE-TAB
+        run: |
+            . /usr/share/miniconda/bin/activate /usr/share/miniconda/envs/perl-atlas
+            perl simple_validate_magetab.pl annotated-projects/PXD000288/PXD000288.idf.tsv
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: sdrf-picture
+          path: graph_file_sdrf.png
+          retention-days: 5
+#  mage_tab_expression_atlas_validations:
+#    runs-on: ubuntu-latest
+#    steps:
+#      - uses: actions/checkout@v2
+#      - name: Setup conda
+#        uses: s-weigand/setup-conda@v1
+#        with:
+#          update-conda: true
+#          conda-channels: anaconda, bioconda, conda-forge
+#      - name: Install perl-atlas-modules
+#        run: conda create -n perl-atlas -c ebi-gene-expression-group perl-atlas-modules
+#      - name: Validate Expression Atlas MAGE-TAB
+#        id: atlas
+#        run: |
+#          . /usr/share/miniconda/bin/activate /usr/share/miniconda/envs/perl-atlas
+#          perl validate_magetab.pl -i annotated-projects/PXD000288/PXD000288.idf.tsv -x -c -v
+#          RSPEC_RESULT=$?
+#          exit 0
+
```

Comment:
In the future, we would like to work with Expression Atlas to make the datasets compatible with them. However, this is not possible yet because they need to update some part of their validator @pcm32